azkaban-common/src/main/java/azkaban/AzkabanCommonModuleConfig.java

```diff
@@ -52,8 +52,12 @@ public AzkabanCommonModuleConfig(final Props props) {
     this.storageImplementation = props.getString(AZKABAN_STORAGE_TYPE, this.storageImplementation);
     this.localStorageBaseDirPath = props
         .getString(AZKABAN_STORAGE_LOCAL_BASEDIR, this.localStorageBaseDirPath);
-    this.hdfsRootUri = props.get(AZKABAN_STORAGE_HDFS_ROOT_URI) != null ? props
-        .getUri(AZKABAN_STORAGE_HDFS_ROOT_URI) : null;
+    this.hdfsProjectRootUri = props.get(AZKABAN_STORAGE_HDFS_PROJECT_ROOT_URI) != null ? props
+        .getUri(AZKABAN_STORAGE_HDFS_PROJECT_ROOT_URI, true) : null;
+    this.cacheDependencyRootUri = props.get(AZKABAN_STORAGE_CACHE_DEPENDENCY_ROOT_URI) != null ? props
+        .getUri(AZKABAN_STORAGE_CACHE_DEPENDENCY_ROOT_URI, true) : null;
+    this.originDependencyRootUri = props.get(AZKABAN_STORAGE_ORIGIN_DEPENDENCY_ROOT_URI) != null ? props
```

Comment:
Can you use `azkaban.utils.Props#getUri(java.lang.String, java.net.URI)` directly?